crates/outbound-networking/src/lib.rs

```diff
@@ -30,6 +30,43 @@ pub fn check_url(url: &str, scheme: &str, allowed_hosts: &AllowedHostsConfig) ->
     is_allowed
 }
 
+#[derive(PartialEq, Debug, Clone)]
+pub enum TemplateableAllowedHost {
+    Literal(AllowedHostConfig),
+    Template(spin_expressions::Template),
+}
+
+impl TemplateableAllowedHost {
+    pub fn parse(pattern: impl Into<String>) -> anyhow::Result<Self> {
+        let pattern = pattern.into();
+        let template = spin_expressions::Template::new(pattern.clone())?;
+        if template.is_literal() {
+            AllowedHostConfig::parse(pattern).map(Self::Literal)
+        } else {
+            Ok(Self::Template(template))
+        }
+    }
+
+    fn snapshot(&self, resolver: &spin_expressions::Resolver) -> anyhow::Result<AllowedHostConfig> {
+        match self {
+            Self::Literal(host) => Ok(host.clone()),
+            Self::Template(template) => {
+                let resolved = futures::executor::block_on(resolver.resolve_template(template))?;
```

Comment:
Ah I hadn't noticed this line before in my haste to suggest that all this code go away. This line would also cause the thread to block...